analyzer/src/main/kotlin/curation/Sw360PackageCurationProvider.kt

```diff
@@ -0,0 +1,133 @@
+import org.ossreviewtoolkit.model.jsonMapper
+
+/**
+ * A [PackageCurationProvider] for curated package meta-data from the configured SW360 instance using the REST API.
+ */
+class Sw360PackageCurationProvider(sw360Configuration: Sw360StorageConfiguration?) : PackageCurationProvider {
+
+    init {
+        requireNotNull(sw360Configuration) {
+            "SW360 is not configured. To use curations from SW360, configure it in the ORT configuration file."
+        }
+    }
+
+    private val sw360Connection = createSw360Connection(
+        sw360Configuration!!,
+        jsonMapper.configure(DeserializationFeature.FAIL_ON_UNKNOWN_PROPERTIES, false)
+    )
+
+    override fun getCurationsFor(pkgId: Identifier): List<PackageCuration> {
+        val name = listOfNotNull(pkgId.namespace, pkgId.name).joinToString("/")
+        val sw360ReleaseClient = sw360Connection.releaseAdapter
+
+        return sw360ReleaseClient.getSparseReleaseByNameAndVersion(name, pkgId.version)
+            .flatMap { sw360ReleaseClient.enrichSparseRelease(it) }
+            .filter { it.sw360ClearingState == SW360ClearingState.APPROVED }
+            .map { sw360Release ->
+                listOf(
+                    PackageCuration(
+                        id = pkgId,
+                        data = PackageCurationData(
+                            declaredLicenses = sw360Release
+                                .embedded
+                                ?.licenses
+                                ?.filterNotNull()
+                                ?.map { license -> license.shortName }
+                                ?.toSortedSet() ?: sortedSetOf(),
+                            homepageUrl = getHomepageOfRelease(sw360Release) ?: "",
+                            binaryArtifact = getAttachmentAsRemoteArtifact(sw360Release, SW360AttachmentType.BINARY)
+                                ?: RemoteArtifact.EMPTY,
+                            sourceArtifact = getAttachmentAsRemoteArtifact(sw360Release, SW360AttachmentType.SOURCE)
+                                ?: RemoteArtifact.EMPTY,
+                            vcs = null,
+                            comment = "Provided by SW360."
+                        )
+                    )
+                )
+            }
+            .orElse(emptyList())
+    }
+
+    private fun getAttachmentAsRemoteArtifact(release: SW360Release, type: SW360AttachmentType): RemoteArtifact? =
+        release
+            .embedded
+            ?.attachments
+            .orEmpty()
+            .filter { it.attachmentType == type }
+            .takeIf { it.size == 1 }
+            ?.let { it ->
+                val attachment = it[0]
+                return RemoteArtifact(
+                    url = attachment.links.self.href,
+                    hash = Hash(
+                        value = attachment.sha1,
+                        algorithm = HashAlgorithm.SHA1
+                    )
+                )
+            }
```

Comment:
How about

        release.embedded?.attachments.orEmpty().singleOrNull { it.attachmentType == type }?.let { attachment ->
            return RemoteArtifact(
                url = attachment.links.self.href,
                hash = Hash(
                    value = attachment.sha1,
                    algorithm = HashAlgorithm.SHA1
                )
            )
        }